include/nighthawk/common/nighthawk_sink_client.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include "envoy/common/pure.h"
+
+#include "external/envoy/source/common/common/statusor.h"
+#include "external/envoy/source/common/protobuf/protobuf.h"
+
+#include "api/client/options.pb.h"
+#include "api/client/service.grpc.pb.h"
+
+namespace Nighthawk {
+
+/**
+ * Interface of a gRPC sink service client.
+ */
+class NighthawkSinkClient {
+public:
+  virtual ~NighthawkSinkClient() = default;
+
+  /**
+   * @brief Store an execution response.
+   *
+   * @param nighthawk_sink_stub Used to open a channel to the sink service.
+   * @param store_execution_request Provide the message that the sink should store.
+   * @return absl::StatusOr<nighthawk::client::StoreExecutionResponse>
+   */
+  virtual absl::StatusOr<nighthawk::client::StoreExecutionResponse> StoreExecutionResponseStream(
```

Comment:
Why do we need to abstract the client behind another interface? Are we going to have multiple implementations?

If it is just for testing, we could do with just one implementation that has virtual methods? The additional layer of indirection comes with its cost, so trying to understand if we have a good reason to pay it.

Reply:
This makes it easy to create a mock for it; so if `ProcessImpl` starts using this further down the road, this comes in handy when we want to unit test the interaction to ensure that looks like we expect it to. But I think you are right in that chances are pretty slim that we'll see >1  implementations for this :-)
 
If you have a concern about perf aspects, imho that only weighs slightly here since this will not be something that will be repeatedly consumed at a frequency where this would become significant; but I can see that there's potential overhead in maintenance of the interface layer.. But here there's also a consistency argument: other similar code does this too.

Weighing everything, I'm personally slightly leaning towards leaving this as is; but I'll ping you on Slack to discuss some more, as changing this is also not something that I'd strongly disagree with.

Reply:
A class with virtual methods should make it as easy to create mocks. The interface tends to only be really needed when there are multiple concrete implementations. That is also how a reader of the code "reads" the existence of a virtual interface. Having it just for mocking may be considered too big of a hammer.

With that said, we don't need to block this PR on it, it is something we can start considering and applying on future code. Let's proceed as is for expediency.